src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -11,47 +11,33 @@
 import java.io.IOException;
 import java.nio.file.Paths;
 
-import org.mapdb.*;
+import org.mapdb.DB;
+import org.mapdb.DBMaker;
+import org.mapdb.BTreeMap;
+import org.mapdb.Serializer;
+
 import com.google.gson.Gson;
 import java.util.Map;
 import java.util.HashMap;
 
 @WebServlet("/recommendation")
 public class RecommendationServlet extends HttpServlet {
-  private DB db;
-  private BTreeMap<String, String[]> queryNearestNeighbors;
-
-  @Override
-  public void init() {
-    String path = Paths.get("").toAbsolutePath().toString() + "/word2vec.db";
-    db = DBMaker.fileDB(path).make();
-    queryNearestNeighbors =
-        db.treeMap("Main")
-            .keySerializer(Serializer.STRING)
-            .valueSerializer(Serializer.JAVA)
-            .createOrOpen();
-  }
-
-  @Override
-  public void destroy() {
-    db.close();
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
-
     UserService userService = UserServiceFactory.getUserService();
     if (!userService.isUserLoggedIn()) {
-      Map<String, String> jsonErrorInfo;
-      jsonErrorInfo = ResponseSerializer.getErrorJson("User not logged in");
-      response.setContentType("application/json;");
-      response.getWriter().println(new Gson().toJson(jsonErrorInfo));
-      return;
+      ResponseSerializer.sendErrorJson(response, "User not logged in");
     }
 
     String queryWord = request.getParameter("queryWord").toLowerCase();
     int numOfWordsRequested = Integer.parseInt(request.getParameter("numOfWordsRequested"));
+    checkforWordRequestedBound(response, numOfWordsRequested);
+
+    // Ensures db is opened in read only to avoid data perturbation
+    String path = Paths.get("").toAbsolutePath().toString() + "/word2vec.db";
+    DB db = DBMaker.fileDB(path).readOnly().make();
```

Comment:
Why was this removed from init()?

Reply:
creating the db needs to be within the same local text of opening the db file. I can't put this in the init() because after the second call to the servlet, it throws an error because the db was closed() and I need to make the db again.